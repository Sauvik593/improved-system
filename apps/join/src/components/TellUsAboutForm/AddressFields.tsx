import { BaseInput } from '@components/common/FormInputs/BaseInput';
import { useTranslation } from 'next-i18next';
import type { UseFormRegister, FieldErrors } from 'react-hook-form';
import type { TellUsAboutFormProps } from './submitForm';

interface Props {
  register: UseFormRegister<TellUsAboutFormProps>;
  errors: FieldErrors<TellUsAboutFormProps>;
}

export const AddressFields = ({ errors, register }: Props) => {
  const { t } = useTranslation('common');
  return (
    <section className="my-4 grid grid-rows-none gap-y-3 lg:grid-cols-2 lg:gap-x-6">
      <div className="lg:col-span-2">
        <BaseInput
          name={'address'}
          label={t('tell_us_about.form.address.label')}
          register={register}
          error={errors.address}
          testId="address"
        />
      </div>

      <BaseInput
        name={'city'}
        label={'Town'}
        register={register}
        error={errors.city}
        testId="address-town"
      />

      <BaseInput
        name={'state'}
        label={t('tell_us_about.form.region.label')}
        register={register}
        error={errors.state}
        testId="address-region"
      />

      <BaseInput
        name={'zip'}
        label={t('tell_us_about.form.postcode.label')}
        register={register}
        error={errors.zip}
        testId="address-postcode"
      />

      <BaseInput
        name={'country'}
        label={t('tell_us_about.form.country.label')}
        register={register}
        error={errors.country}
        testId="address-country"
      />
    </section>
  );
};
