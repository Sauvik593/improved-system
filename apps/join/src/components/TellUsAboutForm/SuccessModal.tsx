import React from 'react';
import NextImage from 'next/image';
import { useTranslation } from 'next-i18next';

import { Dialog, Transition } from '@headlessui/react';

import { Close } from '@kyero/icons';
import { getAssetsUrl } from '@helpers/assetsUrl';

interface Props {
  email: string;
  name: string;
  active: boolean;
  onClose: () => void;
}

export const SuccessModal = ({ active, onClose, email, name }: Props) => {
  const { t } = useTranslation('common');
  return (
    <Transition show={active}>
      <Dialog onClose={onClose}>
        <Transition.Child
          as={'div'}
          enter="ease-out duration-100"
          enterFrom="opacity-0"
          enterTo="opacity-100"
          leave="ease-in duration-100 delay-100"
          leaveFrom="opacity-100"
          leaveTo="opacity-0"
          className="fixed inset-0 z-10 flex items-center justify-center overflow-y-auto"
        >
          <Dialog.Backdrop
            className="fixed inset-0 bg-black/40"
            aria-hidden="true"
            onClick={onClose}
          />
        </Transition.Child>
        <Transition.Child
          as={'div'}
          enter="ease-out duration-100 delay-100"
          enterFrom="opacity-0 translate-y-12"
          enterTo="opacity-100 translate-y-0"
          leave="ease-in duration-100"
          leaveFrom="opacity-50 translate-y-0"
          leaveTo="opacity-0 translate-y-12"
          className="fixed inset-0 z-10 flex items-center justify-center overflow-y-auto"
        >
          <Dialog.Panel
            className="relative z-0 h-auto w-full rounded-xl bg-white p-6 md:mt-0 md:max-w-2xl"
            data-testid="success-modal"
          >
            <button className="absolute top-4 right-4 p-2" title="Close Modal" onClick={onClose}>
              <Close ariaHidden />
            </button>
            <section className="text-center">
              <figure className="mx-auto -mt-20 flex w-auto items-center justify-center">
                <NextImage
                  src={getAssetsUrl('/images/home-page/hero-banner/banner-figure.png')}
                  alt={t('tell_us_about.success.image_alt')}
                  width={312}
                  height={272}
                />
              </figure>
              <h3 className="text-h-4 md:text-h-3 mb-5 font-bold">
                {t('tell_us_about.success.title')}
              </h3>
              <p className="text-p-2 mb-4">{t('tell_us_about.success.message', { name })}</p>
              <div
                className="text-p-2 mx-auto lg:w-3/4"
                dangerouslySetInnerHTML={{
                  __html: t('tell_us_about.success.email_message', { email }),
                }}
              />
            </section>
          </Dialog.Panel>
        </Transition.Child>
      </Dialog>
    </Transition>
  );
};
