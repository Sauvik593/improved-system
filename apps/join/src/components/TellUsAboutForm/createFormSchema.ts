import * as yup from 'yup';
import { TFunction } from 'i18next';

export const createFormSchema = (t: TFunction) =>
  yup.object().shape({
    firstName: yup.string().required(t('form.validations.first_name')),
    lastName: yup.string().required(t('form.validations.last_name')),
    email: yup
      .string()
      .required(t('form.validations.email.required'))
      .email(t('form.validations.email.format')),
    company: yup.string().required(t('form.validations.business_name')),
    companyWebsite: yup.string().optional(),
    phone: yup
      .string()
      .required(t('form.validations.phone_number.required'))
      .min(8, t('form.validations.phone_number.format')),
    vatNumber: yup.string().optional(),
    legalName: yup.string().optional(),
    addressStreet: yup.string().optional(),
    adressCity: yup.string().optional(),
    addressRegion: yup.string().optional(),
    addressPostcode: yup.string().optional(),
    addressCountry: yup.string().optional(),
  });
