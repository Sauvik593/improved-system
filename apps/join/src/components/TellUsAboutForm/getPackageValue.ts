import { tabPanelKeys, listingsNumber, type PlanKey } from '@components/Pricing/PlansSeed';

export const getPackageValue = (listings: string, duration: PlanKey) => {
  if (!tabPanelKeys.includes(duration as PlanKey) || !listingsNumber.includes(listings)) {
    return '200_prime_listings_1_month';
  }
  return `${listings}_prime_listings_${duration}`;
};
