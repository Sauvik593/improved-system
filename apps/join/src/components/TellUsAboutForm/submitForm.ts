import axios from 'axios';

import { HubspotFormResponse } from '@lib/api/hubspot/HubspotForm';
import { getApiUrl } from '@helpers/assetsUrl';

export interface TellUsAboutFormProps {
  firstName: string;
  lastName: string;
  email: string;
  company: string;
  website: string;
  phone: string;
  defaultNationContact: string;
  hsLanguage: string;
  primePackageSelected: string;
  taxNumberContact?: string;
  legalName?: string;
  address?: string;
  city?: string;
  state?: string;
  zip?: string;
  country?: string;
}

export async function submitForm(payload: TellUsAboutFormProps): Promise<HubspotFormResponse> {
  const result = await axios.post(
    getApiUrl('/hubspot'),
    { ...payload },
    { headers: { type: 'tell_us_about' } },
  );
  return result && result.status === 200 && result.data.result;
}
