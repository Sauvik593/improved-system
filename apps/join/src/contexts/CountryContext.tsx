import { createContext } from 'react';
import type { Country } from '@lib/types';

export interface CountryContextParams {
  countries: Country[];
  country: Country | null;
}

export const CountryContext = createContext<CountryContextParams>({
  countries: [],
  country: null,
});
