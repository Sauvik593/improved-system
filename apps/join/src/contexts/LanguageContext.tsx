import { createContext } from 'react';

export interface LanguageContextParams {
  locale: string;
  locales: string[];
}

export const LanguageContext = createContext<LanguageContextParams>({
  locale: '',
  locales: [],
});
