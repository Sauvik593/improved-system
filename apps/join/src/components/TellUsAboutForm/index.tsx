import React, { useContext, useEffect } from 'react';
import { useTranslation } from 'next-i18next';
import { useForm } from 'react-hook-form';
import { startCase } from 'lodash';
import { createFormSchema } from './createFormSchema';
import { yupResolver } from '@hookform/resolvers/yup';

import { BaseInput } from '@components/common/FormInputs/BaseInput';
import { SuccessModal } from './SuccessModal';
import { TellUsAboutFormProps } from './submitForm';
import { useHubspotForm } from '@hooks/useHubspotForm';
import { submitForm } from './submitForm';
import { ErrorModal } from '../common/ErrorModal';
import { HiddenInput } from '@components/common/FormInputs/HiddenInput';
import { LanguageContext } from '@contexts/LanguageContext';
import { CountryContext } from '@contexts/CountryContext';
import { AddressFields } from './AddressFields';
import { LoadingButton } from '@components/common/LoadingButton';
import type { PlanKey } from '@components/Pricing/PlansSeed';
import { getPackageValue } from './getPackageValue';

export const TellUsAboutForm = ({
  listings,
  duration,
  isLoading,
}: {
  listings: string;
  duration: PlanKey;
  isLoading: boolean;
}) => {
  const { t } = useTranslation('common');
  const { locale } = useContext(LanguageContext);
  const { country } = useContext(CountryContext);
  const { error, setFormState, submit, success } = useHubspotForm({
    formService: () => submitForm(getValues()),
  });
  const {
    handleSubmit,
    getValues,
    register,
    reset,
    formState: { errors, isSubmitting },
  } = useForm<TellUsAboutFormProps>({
    resolver: yupResolver(createFormSchema(t)),
  });

  useEffect(() => {
    if (isLoading) {
      reset();
    }
  }, [isLoading, reset]);

  const onClose = () => {
    setFormState('form');
    reset();
  };

  return (
    <section
      data-testid="form-wrapper"
      className="order-2 max-w-2xl rounded-xl bg-white shadow-sm sm:order-1"
    >
      <form className="text-tile-100 p-6" onSubmit={handleSubmit(submit)} data-testid="form">
        <header className="mb-4">
          <h3 className="text-h-4 lg:text-h-3 mb-2 font-bold">{t('tell_us_about.form.title')}</h3>
          <p className="text-p-2-sm lg:text-p-2">{t('tell_us_about.form.subtitle')}</p>
        </header>
        <section className="mb-4 grid grid-rows-none gap-y-4 lg:grid-cols-2 lg:gap-x-6">
          <BaseInput
            name={'firstName'}
            label={t('tell_us_about.form.first_name.label')}
            register={register}
            error={errors.firstName}
            testId="first-name"
          />

          <BaseInput
            name={'lastName'}
            label={t('tell_us_about.form.last_name.label')}
            register={register}
            error={errors.lastName}
            testId="last-name"
          />
        </section>

        <BaseInput
          name="email"
          label={t('tell_us_about.form.email.label')}
          register={register}
          error={errors.email}
          description={t('tell_us_about.form.email.description')}
          testId="email"
        />

        <p className="text-p-2-sm lg:text-p-2 my-4">{t('tell_us_about.form.business_info')}</p>

        <section className="space-y-4 lg:w-3/4">
          <BaseInput
            name={'company'}
            label={t('tell_us_about.form.company.label')}
            description={t('tell_us_about.form.company.description')}
            register={register}
            error={errors.company}
            testId="company-name"
          />

          <BaseInput
            name={'website'}
            label={t('tell_us_about.form.website.label')}
            description={t('tell_us_about.form.website.description')}
            register={register}
          />

          <BaseInput
            name={'phone'}
            label={t('tell_us_about.form.phone.label')}
            description={t('tell_us_about.form.phone.description')}
            register={register}
            error={errors.phone}
            testId="phone-number"
          />

          <BaseInput
            name={'taxNumberContact'}
            label={t('tell_us_about.form.vat_number.label')}
            description={t('tell_us_about.form.vat_number.description')}
            register={register}
            error={errors.taxNumberContact}
            testId="vat-number"
          />
        </section>
        <AddressFields register={register} errors={errors} />

        <HiddenInput
          name={'primePackageSelected'}
          register={register}
          value={getPackageValue(listings, duration)}
          testId={'prime-package-selected'}
        />
        <HiddenInput
          name={'defaultNationContact'}
          register={register}
          value={startCase(country?.translation_key as string)}
        />
        <HiddenInput name={'hsLanguage'} register={register} value={locale} />
        <footer className="my-4 space-y-3">
          <LoadingButton loading={isSubmitting} label={t('tell_us_about.form.cta.title')} />
        </footer>
      </form>
      <SuccessModal
        active={success}
        onClose={onClose}
        email={getValues('email')}
        name={getValues('firstName')}
      />
      <ErrorModal active={error} onClose={() => setFormState('form')} />
    </section>
  );
};
