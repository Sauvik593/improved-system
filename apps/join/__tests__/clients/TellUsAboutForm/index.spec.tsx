import {
  render,
  screen,
  waitFor,
  getByTestId,
  fireEvent,
  act,
  RenderResult,
} from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';

import { TellUsAboutForm } from '@components/TellUsAboutForm';
import { CountryContext } from '@contexts/CountryContext';
import { LanguageContext } from '@contexts/LanguageContext';
import type { PlanKey } from '@components/Pricing/PlansSeed';

import { setupIntersectionObserverMock } from '../../intersectionObserverMock';
import { countrySpain, mockedCountries } from '../../countriesMock';

interface FormValues {
  email: string;
  phone: string;
  firstName: string;
  lastName: string;
  company: string;
}

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

const server = setupServer(
  rest.post('/new-join/api/hubspot', (_req, res, ctx) => res(ctx.status(200))),
);

const submitForm = async (values: FormValues): Promise<void> => {
  const formWrapper = screen.getByTestId('form-wrapper');
  const emailField = getByTestId(formWrapper, 'email') as HTMLInputElement;
  const phoneNumberField = getByTestId(formWrapper, 'phone-number') as HTMLInputElement;
  const firstNameField = getByTestId(formWrapper, 'first-name') as HTMLInputElement;
  const lastNameField = getByTestId(formWrapper, 'last-name') as HTMLInputElement;
  const businessNameField = getByTestId(formWrapper, 'company-name') as HTMLInputElement;
  const form = getByTestId(formWrapper, 'form') as HTMLFormElement;

  await act(async () => {
    await waitFor(async () => {
      fireEvent.change(emailField, { target: { value: values.email } });
      fireEvent.change(phoneNumberField, { target: { value: values.phone } });
      fireEvent.change(firstNameField, { target: { value: values.firstName } });
      fireEvent.change(lastNameField, { target: { value: values.lastName } });
      fireEvent.change(businessNameField, { target: { value: values.company } });
      fireEvent.submit(form);
    });
  });
};

describe('TellUsAboutForm', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(
      <CountryContext.Provider value={{ country: countrySpain, countries: mockedCountries }}>
        <LanguageContext.Provider value={{ locale: 'en', locales: ['en', 'es', 'pt', 'fr', 'it'] }}>
          <TellUsAboutForm isLoading={false} listings="25" duration="1_month" />
        </LanguageContext.Provider>
      </CountryContext.Provider>,
    );
    setupIntersectionObserverMock();
  });

  beforeAll(() => server.listen());
  afterEach(() => {
    server.resetHandlers();
  });
  afterAll(() => server.close());

  describe('When initially rendered', () => {
    it('displays the form view', () => {
      const successModal = screen.queryByTestId('success-modal');
      const errorModal = screen.queryByTestId('error-modal');

      expect(successModal).not.toBeInTheDocument();
      expect(errorModal).not.toBeInTheDocument();
    });
  });

  describe('When the form is submitted', () => {
    it('displays the form success', async () => {
      await submitForm({
        email: 'test@test.com',
        phone: '1234567890',
        firstName: 'John',
        lastName: 'Doe',
        company: 'ACME INC',
      });

      await waitFor(() => {
        expect(screen.queryByTestId('success-modal')).toBeInTheDocument();
        expect(screen.queryByTestId('error-modal')).not.toBeInTheDocument();
      });
    });
  });

  describe('When the form is submitted but request fails', () => {
    it('displays the error modal', async () => {
      server.use(rest.post('/new-join/api/hubspot', (req, res, ctx) => res(ctx.status(500))));

      await submitForm({
        email: 'test@test.com',
        phone: '1234567890',
        firstName: 'John',
        lastName: 'Doe',
        company: 'ACME INC',
      });

      await waitFor(() => {
        expect(screen.queryByTestId('error-modal')).toBeInTheDocument();
        expect(screen.queryByTestId('success-modal')).not.toBeInTheDocument();
      });
    });
  });
  describe('When form is filled', () => {
    it('shows validation errors', async () => {
      const { asFragment } = renderResult;
      await submitForm({
        email: 'wrong_email',
        phone: '123',
        firstName: '',
        lastName: '',
        company: '',
      });

      await waitFor(() => {
        expect(screen.queryByTestId('first-name-error')).toBeInTheDocument();
        expect(screen.queryByTestId('last-name-error')).toBeInTheDocument();
        expect(screen.queryByTestId('email-error')).toBeInTheDocument();
        expect(screen.queryByTestId('phone-number-error')).toBeInTheDocument();
        expect(screen.queryByTestId('company-name-error')).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});

describe('When wrong params are passed to form', () => {
  it('should fill in backup value', () => {
    render(
      <CountryContext.Provider value={{ country: countrySpain, countries: mockedCountries }}>
        <LanguageContext.Provider value={{ locale: 'en', locales: ['en', 'es', 'pt', 'fr', 'it'] }}>
          <TellUsAboutForm
            isLoading={false}
            listings="400"
            duration={'today' as string as PlanKey}
          />
        </LanguageContext.Provider>
      </CountryContext.Provider>,
    );

    const primePackageSelected = screen.getByTestId('prime-package-selected') as HTMLInputElement;

    expect(primePackageSelected.value).toBe('200_prime_listings_1_month');
  });
});
