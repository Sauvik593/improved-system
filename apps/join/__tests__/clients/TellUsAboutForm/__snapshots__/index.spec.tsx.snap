// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TellUsAboutForm When form is filled shows validation errors 1`] = `
<DocumentFragment>
  <section
    class="order-2 max-w-2xl rounded-xl bg-white shadow-sm sm:order-1"
    data-testid="form-wrapper"
  >
    <form
      class="text-tile-100 p-6"
      data-testid="form"
    >
      <header
        class="mb-4"
      >
        <h3
          class="text-h-4 lg:text-h-3 mb-2 font-bold"
        >
          tell_us_about.form.title
        </h3>
        <p
          class="text-p-2-sm lg:text-p-2"
        >
          tell_us_about.form.subtitle
        </p>
      </header>
      <section
        class="mb-4 grid grid-rows-none gap-y-4 lg:grid-cols-2 lg:gap-x-6"
      >
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="firstName"
          >
            tell_us_about.form.first_name.label
          </label>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            data-testid="first-name"
            name="firstName"
            type="text"
          />
          <p
            class="px-2 text-terracotta-100 mt-2"
            data-testid="first-name-error"
          >
            form.validations.first_name
          </p>
        </div>
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="lastName"
          >
            tell_us_about.form.last_name.label
          </label>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            data-testid="last-name"
            name="lastName"
            type="text"
          />
          <p
            class="px-2 text-terracotta-100 mt-2"
            data-testid="last-name-error"
          >
            form.validations.last_name
          </p>
        </div>
      </section>
      <div>
        <label
          class="text-p-2 md:text-h-6 font-bold text-tile-100"
          for="email"
        >
          tell_us_about.form.email.label
        </label>
        <p
          class="text-p-3-sm lg:text-p-3 text-sierra-night-60 w-full"
        >
          tell_us_about.form.email.description
        </p>
        <input
          class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
          data-testid="email"
          name="email"
          type="text"
        />
        <p
          class="px-2 text-terracotta-100 mt-2"
          data-testid="email-error"
        >
          form.validations.email.format
        </p>
      </div>
      <p
        class="text-p-2-sm lg:text-p-2 my-4"
      >
        tell_us_about.form.business_info
      </p>
      <section
        class="space-y-4 lg:w-3/4"
      >
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="company"
          >
            tell_us_about.form.company.label
          </label>
          <p
            class="text-p-3-sm lg:text-p-3 text-sierra-night-60 w-full"
          >
            tell_us_about.form.company.description
          </p>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            data-testid="company-name"
            name="company"
            type="text"
          />
          <p
            class="px-2 text-terracotta-100 mt-2"
            data-testid="company-name-error"
          >
            form.validations.business_name
          </p>
        </div>
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="website"
          >
            tell_us_about.form.website.label
          </label>
          <p
            class="text-p-3-sm lg:text-p-3 text-sierra-night-60 w-full"
          >
            tell_us_about.form.website.description
          </p>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            name="website"
            type="text"
          />
        </div>
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="phone"
          >
            tell_us_about.form.phone.label
          </label>
          <p
            class="text-p-3-sm lg:text-p-3 text-sierra-night-60 w-full"
          >
            tell_us_about.form.phone.description
          </p>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            data-testid="phone-number"
            name="phone"
            type="text"
          />
          <p
            class="px-2 text-terracotta-100 mt-2"
            data-testid="phone-number-error"
          >
            form.validations.phone_number.format
          </p>
        </div>
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="taxNumberContact"
          >
            tell_us_about.form.vat_number.label
          </label>
          <p
            class="text-p-3-sm lg:text-p-3 text-sierra-night-60 w-full"
          >
            tell_us_about.form.vat_number.description
          </p>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            data-testid="vat-number"
            name="taxNumberContact"
            type="text"
          />
        </div>
      </section>
      <section
        class="my-4 grid grid-rows-none gap-y-3 lg:grid-cols-2 lg:gap-x-6"
      >
        <div
          class="lg:col-span-2"
        >
          <div>
            <label
              class="text-p-2 md:text-h-6 font-bold text-tile-100"
              for="address"
            >
              tell_us_about.form.address.label
            </label>
            <input
              class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
              data-testid="address"
              name="address"
              type="text"
            />
          </div>
        </div>
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="city"
          >
            Town
          </label>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            data-testid="address-town"
            name="city"
            type="text"
          />
        </div>
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="state"
          >
            tell_us_about.form.region.label
          </label>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            data-testid="address-region"
            name="state"
            type="text"
          />
        </div>
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="zip"
          >
            tell_us_about.form.postcode.label
          </label>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            data-testid="address-postcode"
            name="zip"
            type="text"
          />
        </div>
        <div>
          <label
            class="text-p-2 md:text-h-6 font-bold text-tile-100"
            for="country"
          >
            tell_us_about.form.country.label
          </label>
          <input
            class="border-sierra-night-40 text-sierra-night-60 w-full rounded-md border p-2 pl-4 mt-2"
            data-testid="address-country"
            name="country"
            type="text"
          />
        </div>
      </section>
      <input
        data-testid="prime-package-selected"
        name="primePackageSelected"
        type="hidden"
        value="25_prime_listings_1_month"
      />
      <input
        name="defaultNationContact"
        type="hidden"
        value="Spain"
      />
      <input
        name="hsLanguage"
        type="hidden"
        value="en"
      />
      <footer
        class="my-4 space-y-3"
      >
        <button
          class="inline-block w-full rounded-full px-4 text-center font-bold text-white disabled:cursor-not-allowed bg-ocean-100 border-ocean-100 hover:border-ocean-120 hover:bg-ocean-120 focus:border-ocean-120 focus:bg-ocean-120 disabled:bg-ocean-40 disabled:border-ocean-40 border-2 py-3"
        >
          tell_us_about.form.cta.title
        </button>
      </footer>
    </form>
  </section>
</DocumentFragment>
`;
